sway-core/src/asm_generation/finalized_asm.rs

```diff
@@ -192,26 +205,251 @@ fn to_bytecode_mut(
     }
 
     assert_eq!(half_word_ix * 4, offset_to_data_section_in_bytes as usize);
-    assert_eq!(buf.len(), offset_to_data_section_in_bytes as usize);
+    assert_eq!(bytecode.len(), offset_to_data_section_in_bytes as usize);
 
-    let config_offsets = data_section
-        .config_map
+    let named_data_section_entries_offsets = data_section
+        .value_pairs
         .iter()
-        .map(|(name, id)| {
+        .enumerate()
+        .filter(|entry| entry.1.name.is_some())
+        .map(|(id, entry)| {
             (
-                name.clone(),
-                offset_to_data_section_in_bytes + data_section.raw_data_id_to_offset(*id) as u64,
+                entry.name.as_ref().unwrap().clone(),
+                offset_to_data_section_in_bytes
+                    + data_section.raw_data_id_to_offset(id as u32) as u64,
             )
         })
         .collect::<BTreeMap<String, u64>>();
 
     let mut data_section = data_section.serialize_to_bytes();
-
-    buf.append(&mut data_section);
+    bytecode.append(&mut data_section);
 
     CompiledBytecode {
-        bytecode: buf,
-        config_const_offsets: config_offsets,
+        bytecode,
+        named_data_section_entries_offsets,
+    }
+}
+
+// Code to pretty print bytecode
+fn print_reg(r: RegId) -> String {
+    if r == RegId::BAL {
+        "$bal".to_string()
+    } else if r == RegId::CGAS {
+        "$cgas".to_string()
+    } else if r == RegId::ERR {
+        "$err".to_string()
+    } else if r == RegId::FLAG {
+        "$flag".to_string()
+    } else if r == RegId::FP {
+        "$fp".to_string()
+    } else if r == RegId::GGAS {
+        "$ggas".to_string()
+    } else if r == RegId::HP {
+        "$hp".to_string()
+    } else if r == RegId::IS {
+        "$is".to_string()
+    } else if r == RegId::OF {
+        "$of".to_string()
+    } else if r == RegId::ONE {
+        "$one".to_string()
+    } else if r == RegId::PC {
+        "$pc".to_string()
+    } else if r == RegId::RET {
+        "$ret".to_string()
+    } else if r == RegId::RETL {
+        "$retl".to_string()
+    } else if r == RegId::SP {
+        "$sp".to_string()
+    } else if r == RegId::SSP {
+        "$ssp".to_string()
+    } else if r == RegId::WRITABLE {
+        "$writable".to_string()
+    } else if r == RegId::ZERO {
+        "$zero".to_string()
+    } else {
+        format!("R{:?}", r.to_u8())
+    }
```

Comment:
This should be a `match`

Reply:
Good point. Not sure why I went with "ifs".

Reply:
Done.